tools/crm_mon_print.c

```diff
@@ -1010,7 +831,13 @@ print_status(pcmk__output_t *out, pe_working_set_t *data_set,
 
     /* Print failed stonith actions */
     if (is_set(show, mon_show_fence_failed) && is_set(mon_ops, mon_op_fence_history)) {
-        print_failed_stonith_actions(out, stonith_history, mon_ops);
+        for (stonith_history_t *hp = stonith_history; hp; hp = hp->next) {
+            if (hp->state == st_failed) {
+                /* Add a blank line between this section and the one before it. */
+                out->info(out, "%s", "");
+                out->message(out, "failed-fencing-history", hp, is_set(mon_ops, mon_op_fence_full_history));
```

Comment:
Wouldn't you need to "break;" in these blocks to avoid printing out the whole list for each entry?

Reply:
Yes, you're right.  Fixing.